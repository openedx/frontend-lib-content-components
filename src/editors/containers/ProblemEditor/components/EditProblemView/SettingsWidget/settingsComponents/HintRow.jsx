import React from 'react';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import {
  Col, Form, Icon, IconButton, Row,
} from '@edx/paragon';
import { Delete } from '@edx/paragon/icons';
import PropTypes from 'prop-types';
import messages from '../messages';

export const HintRow = ({
  value,
  handleChange,
  handleDelete,
  // inject
  intl,
}) => (
  <>
    <Row>
      <Col xs={10}>
        <Form.Group>
          <Form.Control
            value={value}
            onChange={handleChange}
            floatingLabel={intl.formatMessage(messages.hintInputLabel)}
          />
        </Form.Group>
      </Col>

      <Col xs={2}>
        <IconButton
          src={Delete}
          iconAs={Icon}
          alt={intl.formatMessage(messages.settingsDeleteIconAltText)}
          onClick={handleDelete}
          variant="secondary"
        />
      </Col>
    </Row>
  </>
);

HintRow.propTypes = {
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(HintRow);
