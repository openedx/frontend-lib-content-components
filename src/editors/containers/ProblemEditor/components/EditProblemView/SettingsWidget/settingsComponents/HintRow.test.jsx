import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import { formatMessage } from '../../../../../../../testUtils';
import { HintRow } from './HintRow';

describe('HintRow', () => {
  const props = {
    value: 'hint_1',
    handleChange: jest.fn(),
    handleDelete: jest.fn(),
    id: '0',
    intl: { formatMessage },
  };

  describe('snapshot', () => {
    test('snapshot: renders hints row', () => {
      expect(shallow(<HintRow {...props} />).snapshot).toMatchSnapshot();
    });
  });
});
