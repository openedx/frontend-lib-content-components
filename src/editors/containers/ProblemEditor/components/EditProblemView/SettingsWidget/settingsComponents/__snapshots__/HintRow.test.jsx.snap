// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintRow snapshot snapshot: renders hints row 1`] = `
<div
  className="hintRow d-flex flex-row flex-nowrap justify-content-end"
>
  <div
    className="flex-grow-1"
  >
    <Form.Group>
      <Form.Control
        floatingLabel="Hint"
        onChange={[MockFunction]}
        value="hint_1"
      />
    </Form.Group>
  </div>
  <div>
    <IconButton
      alt="Delete answer"
      iconAs="Icon"
      onClick={[MockFunction]}
      variant="secondary"
    />
  </div>
</div>
`;
